core/src/main/scala/pink/cozydev/protosearch/SearchSchema.scala

```diff
@@ -37,19 +39,8 @@ class SearchSchema[A] private (
 }
 object SearchSchema {
   def apply[A](
-      head: (String, A => String, Analyzer),
-      tail: (String, A => String, Analyzer)*
+      head: (Field, A => String),
```

Comment:
This means making a `SearchSchema` is more verbose, as we can see in other changes, but I think that's fine.
We should avoid coming up with DSL like syntax for now and focus on a "kernel" like layer.
DSLs and other entry points can come later.